src/examples/hotgym_NetworkAPI/hotgym_napi.cpp

```diff
@@ -0,0 +1,132 @@
+/* ---------------------------------------------------------------------
+ * HTM Community Edition of NuPIC
+ * Copyright (C) 2013-2015, Numenta, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses.
+ * --------------------------------------------------------------------- */
+
+#include <htm/engine/Network.hpp>
+#include <htm/utils/Random.hpp>
+#include <htm/utils/Log.hpp>
+#include <fstream>
+
+using namespace htm;
+
+static bool verbose = false;
+#define VERBOSE if (verbose)  std::cout << "    "
+
+
+//this runs as an executable
+
+int main(int argc, char* argv[]) {
+  htm::UInt EPOCHS = 5000;      // number of iterations (calls to encoder/SP/TP compute() )
+  const UInt DIM_INPUT = 1000;  // Width of encoder output
+  const UInt COLS = 2048;       // number of columns in SP, TP
+  const UInt CELLS = 8;         // cells per column in TP
+  Random rnd(42);               // uses fixed seed for deterministic output
+  std::ofstream ofs;
+
+  //  Runtime arguments:  hotgym_napi [epochs [filename]]
+  if(argc >= 2) {
+    EPOCHS = std::stoi(argv[1]);         // number of iterations (default 5000)
+  }
+  if (argc >= 3) {
+    ofs.open(argv[2], std::ios::out);    // output filename (for plotting)
+  }
+
+  try {
+
+    std::cout << "initializing\n";
+
+    Network net;
+
+    // Declare the regions to use
+    std::string encoder_parameters = "{size: " + std::to_string(DIM_INPUT) + ", activeBits: 4, radius: 0.5, seed: 2019 }";
+    std::shared_ptr<Region> region1 =  net.addRegion("region1", "RDSERegion", encoder_parameters);
+    std::shared_ptr<Region> region2a = net.addRegion("region2a", "SPRegion", "{columnCount: " + std::to_string(COLS) + "}");
+    std::shared_ptr<Region> region2b = net.addRegion("region2b", "SPRegion", "{columnCount: " + std::to_string(COLS) + ", globalInhibition: true}");
```

Comment:
regarding parameters, these setings use SPs' etc default params, I think it'd be nice to use: 
- I think the best would be if you also add SensorRegion to show reading a CSV file, so to replicate `hotgym.py` (uses the real hotgym data, also then use its params)

Reply:
Ok, I did not even think to pattern it after hotgym.py.  I assumed it was the same as HelloSPTP.cpp.  I will check that out.